Modules/cPickle.c

```diff
@@ -586,13 +586,15 @@ readline_file(Unpicklerobject *self, char **s)
     while (1) {
         Py_ssize_t bigger;
         char *newbuf;
-        for (; i < (self->buf_size - 1); i++) {
-            if (feof(self->fp) ||
-                (self->buf[i] = getc(self->fp)) == '\n') {
-                self->buf[i + 1] = '\0';
-                *s = self->buf;
-                return i + 1;
-            }
+        int newchar;
+        while (i < (self->buf_size - 1)) {
+            if (feof(self->fp))
```

Comment:
Isn't this redundant with the check on line 593?

Reply:
I don't know, is it? I can never remember the guaranteed semantics of FILE*s, feof() and functions that return EOF (which can signal error). Since the original function used this, I kept this it in.

Reply:
My manpage says `getc()` returns EOF on end-of-file and error, so that's why I believe this is subsumed by the next call.